### DISCUSSION TITLE
How do you make portfolio projects filter by tag buttons?

### DISCUSSION THREAD

**Discussion by annaozola** [gatsbyjs/gatsby#35887] [Help]
I'm trying to make a portfolio projects page where in the projects page every project shows up and I have tag buttons I can click to filter the projects. There's also a tags page where all the tags are displayed.

This is a good example of what I'm trying to make: https://www.w3schools.com/howto/howto_js_portfolio_filter.asp
Here's screenshots of the pages: https://imgur.com/a/AujVbkz

### Issues

1. I don't know how to configure the projects to show up in the projects page and how to add the tag buttons so they would actually work.
2. The tag pages such as /tags/materials/ are empty when I click on the materials tag in /tags/. It should show me the projects with the tag.

I'm quite new to GraphQL and Gatsby so this is really a struggle for me. There are similar templates made for Gatsby that I have already gone through but they use TypeScript and I don't understand it so it didn't help me. I really enjoy working with Gatsby so I really want to resolve this.

**Comment by violy** [comment]
Few questions : 
- How are you sourcing your portfolio projects ?
- Is all your projects on a single page / URL ?
